src/apps/setup.ts

```diff
@@ -21,7 +21,7 @@ export const setupAppFactory = (
     // If not on Glitch or Production, create a smee URL
     if (
       process.env.NODE_ENV !== "production" &&
-      !(process.env.PROJECT_DOMAIN || process.env.WEBHOOK_PROXY_URL)
+      !(process.env.PROJECT_DOMAIN || process.env.WEBHOOK_PROXY_URL || host)
```

Comment:
how is this change related?

Reply:
This change fixes also a setup problem. But this is probably a general problem.

My setup is the following:
- Computer with dns name reachable from the github enterprise server
- Each app has its own docker container
- Each app has a different port

I have the dns name set in the "env.HOST" variable, also "env.PORT" is set. env.NODE_ENV is not set.
No keys or app ids are set since I want to run the setup through the web page of the app (and they should all be set by the setup process).

Without the addition, a smee webhook is created automatically and will then therefore be set on github as the endpoint instead of the real dns name and port. See https://github.com/probot/probot/blob/ba1bfb0ef519244421af5425428b5962179c4744/src/manifest-creation.ts#L50
So the current implementation does not allow non production spin up together with setting the host and this is needed for the setup process to be correct. I worked around it  locally by setting PROJECT_DOMAIN  but that changes the proposed app name in the setup and is therefore not optimal.

Reply:
I can add some documentation examples how I used it, in general this is not really GHES specific.

Actually I'm not sure how this was ever used outside of development (with smee) or together with PROJECT_DOMAIN.
So I presume every body who wanted to have an app running permanently on a real server with dns name probably  (without smee) used the manual setup. This is true for github.com and GHES.

I looked at the code and could only find that HOST is read here: https://github.com/probot/probot/blob/ce9688493029f32a5abdeb3a4fb91d56d00c18e5/src/bin/read-env-options.ts#L14
and it is set here: https://github.com/probot/probot/blob/ce9688493029f32a5abdeb3a4fb91d56d00c18e5/src/run.ts#L27
so I'm confident that host is only set if HOST is set.

I'm actually not sure what HOST is used for in a production environment.
It is used: https://github.com/probot/probot/blob/ce9688493029f32a5abdeb3a4fb91d56d00c18e5/src/server/server.ts#L77 for express listen, and the setup of the app (but this is not working since it is replace with smee at the moment)
Otherwise it is shown to the user here: https://github.com/probot/probot/blob/ce9688493029f32a5abdeb3a4fb91d56d00c18e5/src/apps/setup.ts#L103

A test for the change could be added here:
https://github.com/probot/probot/blob/ce9688493029f32a5abdeb3a4fb91d56d00c18e5/test/manifest-creation.test.ts#L74
or here:
https://github.com/probot/probot/blob/ce9688493029f32a5abdeb3a4fb91d56d00c18e5/test/manifest-creation.test.ts#L126
Would you like me to add a test with HOST set to the manifest or the normal run tests? I would like to get a solution since I have to setup quite a few apps and hope the ui setup is faster then doing it manual :-)

Here is an example of how I use it:
App is in docker with a app.yml, app is in the folder '/app'
An empty file test-app-settings.env
docker-compose.yml (part)
```yaml
testapp:
    image: github-apps/testapp:prod
    container_name: testapp
    volumes:
      - ./test-app-settings.env:/app/.env
    ports:
      - 3001:3001
    restart: unless-stopped
    environment:
      # General github app settings
      - LOG_LEVEL=info
      - PORT=3001
      - HOST=http://DockerTestServer.abc.com
      - PROJECT_DOMAIN=http://DockerTestServer.abc.com:3001
      #- NODE_ENV=production

      # Github server if needed
      # - GHE_HOST=github.abc.com
```

Reply:
Or maybe I'm wrong with my usage of host all together and HOST is only meant to be used for the listening ip on the server?
But then there should be an option to not create the smee proxy in case there is a dns name available?
Maybe it is easier to just add an option NO_SMEE (or something like that) and then https://github.com/probot/probot/blob/ce9688493029f32a5abdeb3a4fb91d56d00c18e5/src/manifest-creation.ts#L50 would create the correct url if it is called correctly.

Reply:
> I'm actually not sure what HOST is used for in a production environment.

I think it's also used in a local environment, some folks set it to `127.0.0.1` to prevent outside requests.

And some deployment platforms set `HOST`, I think Heroku does



> I looked at the code and could only find that HOST is read here:
> 
> https://github.com/probot/probot/blob/ce9688493029f32a5abdeb3a4fb91d56d00c18e5/src/bin/read-env-options.ts#L14

I've seen the same thing. I'm okay with merging it now, worst case we reverse the change if something blows up